google-cloud-firestore/src/main/java/com/google/cloud/firestore/BulkWriter.java

```diff
@@ -167,32 +249,41 @@
    *
    * @param documentReference A reference to the document to be created.
    * @param pojo The POJO that will be used to populate the document contents.
-   * @return An ApiFuture containing the result of the write. Contains an error if the write fails.
+   * @return An ApiFuture containing the result of the write. Contains a {@link BulkWriterException}
+   *     if the write fails.
    */
   @Nonnull
   public ApiFuture<WriteResult> create(
-      @Nonnull DocumentReference documentReference, @Nonnull Object pojo) {
+      @Nonnull final DocumentReference documentReference, @Nonnull final Object pojo) {
     verifyNotClosed();
-    BulkCommitBatch bulkCommitBatch = getEligibleBatch();
-    ApiFuture<WriteResult> future = bulkCommitBatch.create(documentReference, pojo);
-    sendReadyBatches();
-    return future;
+    return executeWrite(
+        documentReference,
+        OperationType.CREATE,
+        new BulkWriterOperationCallback() {
+          public ApiFuture<WriteResult> apply(BulkCommitBatch batch) {
+            return batch.set(documentReference, pojo);
+          }
+        });
   }
 
   /**
    * Delete a document from the database.
    *
    * @param documentReference The DocumentReference to delete.
-   * @return An ApiFuture containing the result of the delete. Contains an error if the delete
-   *     fails.
+   * @return An ApiFuture containing a sentinel value (Timestamp(0)) for the delete operation.
```

Comment:
Can we explain why the sentinel value is returned? (Also, 0 probably isn't the best sentinel value since Jan 1 1970 is a pretty practical date for a lot of data). This I think also differs from the behavior of `DocRef.delete`and `batch.delete` which I believe use the server commit response time.